tool/generator/lib/src/projection/utils.dart

```diff
@@ -557,6 +557,54 @@ String stripGenerics(String name) {
   return name.substring(0, backtickIndex);
 }
 
+/// Sorts [importLines] according to Effective Dart: Style guidelines.
+/// See https://dart.dev/guides/language/effective-dart/style#ordering
+List<String> sortImports(List<String> importLines) {
+  if (importLines.isEmpty) return importLines;
+
+  final dartImports = <String>[];
+  final packageImports = <String>[];
+  final projectImports = <String>[];
+  final projectRelativeImports = <String>[];
```

Comment:
Should these four lines be `Set`s? Presumably we don't want duplicates...

Reply:
`Set` doesn't have a `sort` method so I've used `SplayTreeSet` which sorts them automatically. Thanks for catching this!